test/testtokenize.cpp

```diff
@@ -3107,11 +3106,6 @@ class TestTokenizer : public TestFixture {
         }
     }
 
-    void simplifyExternC() {
-        ASSERT_EQUALS("int foo ( ) ;", tokenizeAndStringify("extern \"C\" int foo();"));
```

Comment:
There will be FPs with your fix for such declarations, right?

    extern "C" void foo(const struct S s);

.. because there is no extern C scope anywhere here.

It feels unfortunate to keep such declaration `extern "C" int foo();` as is in the Tokenizer. We have code here and there that parse declarations and that code is not expecting to have a string literal in the middle of the declaration. I think we have 2 options:
 * Remove the `extern "C"` tokens and add a new attribute in `Token`. Set attribute for relevant tokens.
 * Merge `extern "C"` into one token. Since it starts with `e` it will be seen as some typename.

If we remove `extern "C"` and add an attribute... I think it's debatable if we should keep the externC scope in SymbolDatabase also. You will be able to fix the FP by using the attribute instead. You made tweaks to handle typedef and function lookups but I have the feeling that there are more lookups that are not handled yet.

Reply:
> Remove the extern "C" tokens and add a new attribute in Token.

Oh, good point.  I just reimplemented it that way, which make this change simpler and less error-prone.  And I added a few more tests based on your comment above.